stan/math/opencl/opencl_context.hpp

```diff
@@ -144,7 +146,7 @@ class opencl_context_base {
   std::string device_name_;          // The name of OpenCL device
   size_t max_thread_block_size_;  // The maximum size of a block of workers on
                                   // the device
-
+  bool block_transfer;            // Whether to use out of order execution.
```

Comment:
I don't see how this is connected to any blocks. I think a better name for this variable would be `in_order_` or `use_asnyc_transfers_` or something like that.